src/main/java/core/basesyntax/service/operation/BalanceOperation.java

```
@@ -0,0 +1,18 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.storage.Storage;
+
+public class BalanceOperation implements OperationHandler {
+    private final Storage storage;
+
+    public BalanceOperation(Storage storage) {
+        this.storage = storage;
```

Constructor stores the provided Storage reference without validation. Validate the `storage` parameter (non-null) and throw an informative FruitShopException if it is null to fail fast and preserve encapsulation (the checklist requires anticipating invalid inputs and following encapsulation). Example message: "Storage is null in BalanceOperation constructor". See the project guidance: "Remember that your code will need to be tested, so try to anticipate and handle all invalid input data in advance."